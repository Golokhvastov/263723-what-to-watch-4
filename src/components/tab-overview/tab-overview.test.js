import React from "react";
import renderer from "react-test-renderer";
import TabOverview from "./tab-overview.jsx";

const mocks = {
  movie: {
    title: `Test1`,
    previewImage: `test1.jpg`,
    genre: `Test genre`,
    year: 1999,
    rating: {
      score: 5.5,
      votes: 110
    },
    descriptions: [
      `Test description1`,
      `Test description2`,
    ],
    director: `Test director`,
    starring: [
      `Test starring 1`,
      `Test starring 2`,
      `Test starring 3`,
      `Test starring 4`,
    ],
    runTime: 120,
    preview: `Test4`,
    reviews: [
      {
        rating: 7.9,
        date: `December 10, 2010`,
        author: `Test5`,
        text: `Test5`
      },
      {
        rating: 8.0,
        date: `December 20, 2020`,
        author: `Test6`,
        text: `Test6`
      },
      {
        rating: 9.0,
        date: `December 30, 2030`,
        author: `Test7`,
        text: `Test7`
      }
    ]
  }
};

it(`TabOverview render correctly`, () => {
  const tree = renderer
    .create(
        <TabOverview
          movie={mocks.movie}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
