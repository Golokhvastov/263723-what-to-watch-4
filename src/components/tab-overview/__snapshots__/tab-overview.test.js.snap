// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabOverview render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      5.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        110
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Test description1
    </p>
    <p>
      Test description2
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Test director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Test starring 1, Test starring 2, Test starring 3, Test starring 4
         and other
      </strong>
    </p>
  </div>,
]
`;
